test/typescript/terraform-cloud/test.ts

```diff
@@ -60,10 +60,34 @@ describe("full integration test", () => {
     });
 
     process.env.TF_EXECUTE_LOCAL = "true";
-    driver.deploy();
+    driver.deploy("source-stack");
     process.env.TF_EXECUTE_LOCAL = undefined;
-    driver.deploy();
+    driver.deploy("source-stack");
 
     await client.Workspaces.deleteByName(orgName, workspaceName);
   });
+
+  withAuth("deploy locally and then in Terraform Cloud", async () => {
+    const client = new TerraformCloud(TERRAFORM_CLOUD_TOKEN);
+
+    await client.Workspaces.create(orgName, {
+      data: {
+        attributes: {
+          name: workspaceName,
+          executionMode: "remote",
+          terraformVersion: TERRAFORM_VERSION,
+        },
+        type: "workspaces",
+      },
+    });
+
+    driver.deploy("source-stack");
+    driver.deploy("consumer-stack");
+
+    await client.Workspaces.deleteByName(orgName, workspaceName);
```

Comment:
Maybe add a note that only source stack is configured to deploy to Terraform Cloud?
For full Terraform Cloud support (both stacks deployed in separate TFC workspaces) we need at least docs / guidance for users to inform them that they have to set the correct permissions to the TFC workspaces if they are using remote execution – else the consumer workspace can't access the state of the source workspace I think.
Ideally the cdktf-cli could read the stack dependencies and check this in TFC and offer to setup the required permissions. But that's definitely something for the future 😁